elm/src/main/scala/io/scalac/elm/state/ElmWallet.scala

```diff
@@ -81,16 +83,29 @@ case class ElmWallet(secret: PrivateKey25519 = generateSecret(),
   override def historyTransactions: Seq[WalletTransaction[PublicKey25519Proposition, ElmTransaction]] = ???
 
   override def scanPersistent(modifier: ElmBlock): ElmWallet = {
-    // increase balance by confirmed outputs
     val transactions = modifier.transactions.getOrElse(Nil)
 
-    transactions.foldLeft(this) { (w, tx) =>
-      val outs = tx.outputs.filter(_.proposition.pubKeyBytes.key == pubKeyBytes)
-      val increasedOutputs = outs.foldLeft(chainTxOutputs)((m, o) => m + (o.id.key -> TimedTxOutput(o, tx.timestamp)))
-      val sum = outs.map(_.value).sum
-
+    // increase balance by confirmed outputs
+    val increasedWallet = {
```

Comment:
I think that we should also count `decressedWallet` for transaction that spends money from wallet that comes from other nodes